src/server/handlers/launchpad.js

```diff
@@ -340,6 +340,47 @@ export const internalFindSnap = async (repositoryUrl) => {
   });
 };
 
+const internalFindSnapsByPrefix = (urlPrefix) => {
+  const username = conf.get('LP_API_USERNAME');
+
+  return getLaunchpad().named_get('/+snaps', 'findByURLPrefix', {
```

Comment:
I guess it might be useful to memcache the responses here (like we do with finding snap by URL)?